### DISCUSSION TITLE
Docker registry-mirrors

### DISCUSSION THREAD

**Discussion by abreudenis** [containrrr/watchtower#1675] [Q&A] [answered]
Hi,

I'm using Watchtower in a host that configs Docker to use registry-mirrors:
```
$ cat /etc/docker/daemon.json
{
    "registry-mirrors": [
        "https://nexus.mycompany.com:5005/"
    ]
}
```
This setting point to a Nexus proxy repo, and due this, our private images are referenced like this: `group/subgroup/project:tag`

However, I don't think Watchtower is aware of this setting because of these log messages.

- Parsing image ref" host=index.docker.io
- Building challenge URL" URL="https://index.docker.io/v2/
- Got response to challenge request" header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\"" status="401 Unauthorized"

Seems Watchtower is looking for new images at docker.io ignoring our registry-mirrors setting.

My Watchtower runs like this:
```
docker run -d --name watchtower -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --interval 300 --debug
```

Any suggest to teach Watchtower to use our registry-mirrors proxy?

Thank you all 💪🏽

**Comment by piksel** [comment, marked as answer]
I would suggest using the full path to the images if they are not available on dockerhub. Right now you are saying that the nexus server is a mirror, but it contains more packages than what it supposedly mirrors. 

Watchtower does not take into account any configuration files on your machine unless they are mounted into the container, but even if you did, there is no support for using mirrors. Docker itself will still use its configuration of course, and the actual pull of the image is done through the docker daemon. 

To avoid running into quota problems by pulling all the time, watchtower first checks if the image has been updated before pulling, this is what fails for you since you do not put your registry in the image tag.

**Comment by abreudenis** [comment]
Thank you, @piksel 🤝
